packages/vite/src/node/config.ts

```diff
@@ -1689,16 +1689,24 @@ async function loadConfigFromBundledFile(
   // with --experimental-loader themselves, we have to do a hack here:
   // write it to disk, load it with native Node ESM, then delete the file.
   if (isESM) {
-    const fileBase = `${fileName}.timestamp-${Date.now()}-${Math.random()
-      .toString(16)
-      .slice(2)}`
-    const fileNameTmp = `${fileBase}.mjs`
-    const fileUrl = `${pathToFileURL(fileBase)}.mjs`
-    await fsp.writeFile(fileNameTmp, bundledCode)
+    const nodeModulesDir = findNearestNodeModules(path.dirname(fileName))
+    if (nodeModulesDir) {
+      await fsp.mkdir(path.resolve(nodeModulesDir, '.vite/configs/'), {
+        recursive: true,
+      })
```

Comment:
At first I thought you wanted to have it always in `node_modules`, independent of the value of `cacheDir`, but with this resolution it will be in the `cacheDir` by default. Was there a reason to want it in `node_modules` specifically? If not, this should use `cacheDir`

Reply:
To achieve that, a dead-lock happens.
We need to load the config to know the `cacheDir`. But we cannot know the `cacheDir` until the config is loaded.

Reply:
Unless `cacheDir` is set in the inline config, we can't really know ahead of time before the config is loaded. It also needs to be in the closest `node_modules`, otherwise dependencies referenced within it can't be properly resolved.

If there's no `node_modules` anywhere, it should fallback to the behaviour before.

Reply:
Ah, I missed that. It is still strange that you change `cacheDir` and some things are still in `node_modules/.vite`. I think we could make `node_modules/.vite-temp` for this?

Reply:
Updated to `node_modules/.vite-temp` 👍